### PR TITLE
Add Kerberos-aware Elasticsearch and integration test to ES output

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR adds an integration test to the Elasticsearch output to check Kerberos authentication.

Furthermore, it adds a new element to our testing environment, a Kerberos-aware Elasticsearch instance named `elasticsearch_kerberos.elastic`.

## Why is it important?

To make sure Kerberos authentication is working.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: libbeat/esleg/eslegtest/util.go [hunk 1]
```diff
@@ -64,6 +64,11 @@ func GetEsHost() string {
 	return getEnv("ES_HOST", ElasticsearchDefaultHost)
 }
 
+// GetEsHost returns the Elasticsearch testing host.
```
Comment (ycombinator) [inline libbeat/esleg/eslegtest/util.go:67]: Typo: update the name of the function in the comment to match.

File: libbeat/docker-compose.yml [hunk 2]
```diff
@@ -42,14 +43,15 @@ services:
...
-      elasticsearch:            { condition: service_healthy }
-      elasticsearch_monitoring: { condition: service_healthy }
-      elasticsearchssl:         { condition: service_healthy }
-      logstash:                 { condition: service_healthy }
-      kafka:                    { condition: service_healthy }
-      redis:                    { condition: service_healthy }
-      sredis:                   { condition: service_healthy }
-      kibana:                   { condition: service_healthy }
+      elasticsearch:                  { condition: service_healthy }
+      elasticsearch_kerberos.elastic: { condition: service_healthy }
```
Comment (ycombinator) [inline libbeat/docker-compose.yml:47]: Nit: just name this `elasticsearch_kerberos` (unless the `.elastic` suffix is somehow significant)?

Comment (kvch) [reply libbeat/docker-compose.yml:47]: It is needed because it is part of the ELASTIC Kerberos realm.

**Comment by andresrc** [comment]
I think some AD environment would still be needed. I don't know if we would be able to use for automated testing but just for support scenarios I think it would be very useful. What options do we have here?

**Pull request merged**